### PR TITLE
Toxic Exposure 1.0: Various production tester flag fixes

### PR DESCRIPTION
**Pull request by sethdarragile6**
This PR fixes issues with `disability_compensation_production_tester` flag:
- the log message for the 4142 bypass
- all Flipper calls with `User.find()` (found to be unreliable after further testing)
- more backup submission trigger fixes

## Summary

- *This work is behind a feature toggle (flipper): YES*
- *(Summarize the changes that have been made to the platform)*
- *(If bug, how to reproduce)*
- *(What is the solution, why is this the solution?)*
- *(Which team do you work for, does your team own the maintenance of this component?)*
- *(If introducing a flipper, what is the success criteria being targeted?)*

## Related issue(s)

- *Link to ticket created in va.gov-team repo OR screenshot of Jira ticket if your team uses Jira*
[#90091](https://app.zenhub.com/workspaces/disability-benefits-experience-team-1-63dbdb0a401c4400119d3a44/issues/gh/department-of-veterans-affairs/va.gov-team/90091)
- *Link to previous change of the code/bug (if applicable)*
- *Link to epic if not included in ticket*

## Testing done

- [ ] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by aurora-a-k-a-lightning — APPROVED** [review]

File: app/models/form526_submission.rb [hunk 1]
```diff
@@ -490,8 +491,8 @@ def last_remediation
   private
 
   def conditionally_submit_form_4142
-    if Flipper.enabled?(:disability_compensation_production_tester, User.find(user_uuid))
-      Rails.logger.info("submit_form_4142 call skipped for submission #{submission.id}")
+    if Flipper.enabled?(:disability_compensation_production_tester, OpenStruct.new({ flipper_id: user_uuid }))
+      Rails.logger.info("submit_form_4142 call skipped for submission #{id}")
```
Comment (tommasina-va) [inline app/models/form526_submission.rb:495]: is `#{id}` supposed to be `#{submission.id}`?

Comment (sethdarragile6) [reply app/models/form526_submission.rb:495]: haha NOPE! That was our original mistake, and it was throwing an exception that prevented the logging entirely. We're already in the context of the submission (just as `user_uuid` is here).

**Review by aurora-a-k-a-lightning — APPROVED** [review]
ship this thing! 🚢

File: app/models/form526_submission.rb (hunk 1 above)
Comment (tommasina-va) [reply app/models/form526_submission.rb:495]: Ah-ha! Sounds good. 👍🏻

**Review by tommasina-va — APPROVED** [review]
LGTM!

**Pull request merged**